crates/ra_syntax/src/ast.rs

```diff
@@ -664,6 +664,65 @@ impl LiteralExpr {
     }
 }
 
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub struct FieldPat {
+    pub ident: SmolStr,
+    // FIXME: could we use a regular reference?
+    pub pat: TreeArc<Pat>,
+}
+
+impl FieldPatList {
+    // TODO: try returning an iterator?
+    // FIXME: shouldnt the parser do this? :o
```

Comment:
yes, it seems to me like the parser should create a `FIELD_PAT` node between the `FIELD_PAT_LIST` and the field identifiers and patterns, which it currently doesn't seem to. @matklad I guess this should be fixed in the parser, right? Or is there a reason for this?

Reply:
Yep, this should be fixed in the parser:

* add `FIELD_PAT` node to the `grammar.ron`
* change the corresponding parser funciton in the grammar to wrap contents of the field into this new node
* add its methods to `ast` section of `grammar.ron`
* run `cargo gen-syntax` & `cargo gen-tests`